src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,133 @@
+package parser;
+
+import command.Command;
+import task.*;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a parser that parses the user input and converts it into a task object.
+ * The parser also converts the date and time strings into LocalDate and LocalTime objects.
+ */
+public class Parser {
+    /**
+     * Parses the user input and returns a task object.
+     *
+     * @param taskType The type of task to be created.
+     * @param command The command object containing the user input.
+     * @return The task object created from the user input.
+     * @throws InvalidTaskException If the user input is invalid.
+     */
+    public static Task parseTask(Task.TYPE taskType, Command command) throws InvalidTaskException {
+        try {
+            switch (taskType) {
+                case TODO: {
```

Comment:
As stated in the Coding Standard, the content of ``case`` shouldn't be inside Egyptian braces, right?